src/ui/MainMenu.cpp

```diff
@@ -3985,25 +4166,1324 @@ namespace MainMenu {
+						if (newRemoteProductId && EOS.P2PConnectionInfo.isPeerIndexed(newRemoteProductId)) {
+							if (EOS.P2PConnectionInfo.getIndexFromPeerId(newRemoteProductId) >= 0) {
+								// we've already accepted this player. NEXT!
+								skipJoin = true;
+							}
+						}
+						if (skipJoin) {
+							continue;
+						}
+#endif // USE_EOS
+					}
+				}
+
+				// process incoming join request
+				NetworkingLobbyJoinRequestResult result = lobbyPlayerJoinRequest(playerNum);
+
+				// finalize connections for Steamworks / EOS
+				if (result == NetworkingLobbyJoinRequestResult::NET_LOBBY_JOIN_P2P_FAILURE) {
+					if (LobbyHandler.getP2PType() == LobbyHandler_t::LobbyServiceType::LOBBY_STEAM) {
+#ifdef STEAMWORKS
+						for (int responses = 0; responses < 5; ++responses) {
+							SteamNetworking()->SendP2PPacket(newSteamID, net_packet->data, net_packet->len, k_EP2PSendReliable, 0);
+							SDL_Delay(5);
+						}
+#endif // STEAMWORKS
+					} else if ( LobbyHandler.getP2PType() == LobbyHandler_t::LobbyServiceType::LOBBY_CROSSPLAY ) {
+#ifdef USE_EOS
+						for ( int responses = 0; responses < 5; ++responses ) {
+							EOS.SendMessageP2P(newRemoteProductId, net_packet->data, net_packet->len);
+							SDL_Delay(5);
+						}
+#endif // USE_EOS
+					}
+				} else if ( result == NetworkingLobbyJoinRequestResult::NET_LOBBY_JOIN_P2P_SUCCESS ) {
+					if ( LobbyHandler.getP2PType() == LobbyHandler_t::LobbyServiceType::LOBBY_STEAM ) {
+#ifdef STEAMWORKS
+						if ( steamIDRemote[playerNum - 1] ) {
+							cpp_Free_CSteamID(steamIDRemote[playerNum - 1]);
+						}
+						steamIDRemote[playerNum - 1] = new CSteamID();
+						*static_cast<CSteamID*>(steamIDRemote[playerNum - 1]) = newSteamID;
+						for ( int responses = 0; responses < 5; ++responses ) {
+							SteamNetworking()->SendP2PPacket(*static_cast<CSteamID* >(steamIDRemote[playerNum - 1]), net_packet->data, net_packet->len, k_EP2PSendReliable, 0);
+							SDL_Delay(5);
+						}
+#endif // STEAMWORKS
+					} else if (LobbyHandler.getP2PType() == LobbyHandler_t::LobbyServiceType::LOBBY_CROSSPLAY) {
+#if defined USE_EOS
+						EOS.P2PConnectionInfo.assignPeerIndex(newRemoteProductId, playerNum - 1);
+						for (int responses = 0; responses < 5; ++responses) {
+							EOS.SendMessageP2P(EOS.P2PConnectionInfo.getPeerIdFromIndex(playerNum - 1), net_packet->data, net_packet->len);
+							SDL_Delay(5);
+						}
+#endif
+					}
+				}
+
+				// assume success after this point
+				assert(result == NET_LOBBY_JOIN_DIRECTIP_SUCCESS ||
+				    result == NET_LOBBY_JOIN_P2P_SUCCESS);
+
+
+				// finally, open a player card!
+				if (playerNum >= 1 && playerNum < MAXPLAYERS) {
+			        createReadyStone(playerNum, false, false);
+				}
+
+				continue;
+			}
+
+			// network scan
+			else if (packetId == 'SCAN') {
+			    if (directConnect) {
+			        char hostname[256] = { '\0' };
+			        (void)gethostname(hostname, sizeof(hostname));
```

Comment:
Note I'm getting missing definition here for `gethostname`

Reply:
ah yes, forgot this is a POSIX function. Fun times.